metadata-ingestion/src/datahub/sdk/dataset.py

```diff
@@ -468,6 +469,7 @@ def __init__(
             env=env,
         )
         super().__init__(urn)
+        self._set_extra_aspects(extra_aspects)
```

Comment:
given we are calling `_set_extra_aspects` for both container and dataset, should we move that responsibility to the parent: `super().__init__(urn, extra_aspects)` ?

Reply:
tbh I'm not super happy with the interface setup right now

Specifically, I'm not sure if set_extra_aspects should be called immediately after the super.init, or at the end of the Dataset/Container init method

Once I have a stronger opinion on it, I'll refactor this stuff